### DISCUSSION TITLE
storing hooks as atoms

### DISCUSSION THREAD

**Discussion by austinwoon** [pmndrs/jotai#1354] [Q&A] [answered]
would it be possible to store an object of state setters returned by hooks as atoms?

Here's what we're doing with the `react-hook-form` `useForm ` hook, but not sure if its safe to do something like this

```tsx
import { useForm } from 'react-hook-form'

export const familyEditFormHookAtom = atom<
  UseFormReturn<FieldValues, object> | undefined
>(undefined)
```

In functional component
```tsx 
const familyFormHook = useForm()

    <Provider
      initialValues={[
        [familyEditFormHookAtom, familyFormHook],
      ]}
    >
```

**Comment by dai-shi** [comment, marked as answer]
Storing hooks in props or contexts or states is not recommended as it can lead violating rules of hooks.
Likewise, storing hooks in atoms (which is essentially contexts+states) is not recommended. (It's possible though.)
